src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -60,9 +114,19 @@ Copyright (c) .NET Foundation. All rights reserved.
     <OutputPath>$(OutputPath)$(RuntimeIdentifier)\</OutputPath>
   </PropertyGroup>
 
-  <Target Name="_CheckRuntimeIdentifier" BeforeTargets="_CheckForInvalidConfigurationAndPlatform">
-    <NetSdkError Condition="'$(RuntimeIdentifier)' == '' and '$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(OutputType)' == 'Exe'"
-                 ResourceName="RuntimeIdentifierMustBeSetForNETFramework" />
+  <!-- 
+    Switch our default .NETFramework CPU architecture choice back to AnyCPU before 
+    compiling the exe if no copy-local native dependencies were resolved from NuGet 
+  -->
+  <Target Name="AdjustDefaultPlatformTargetForNetFrameworkExeWithNoNativeCopyLocalItems"
```

Comment:
How likely is it that there could be something else that consumes the `PlatformTarget` property between when we set it to x86 or x64 based on the defaulted RuntimeIdentifier and when we set it back to AnyCPU because we didn't resolve any native assets?